src/core/thread/time_sync_service.cpp

```diff
@@ -128,6 +134,14 @@ void TimeSync::IncrementTimeSyncSeq(void)
     }
 }
 
+void TimeSync::NotifyTimeSyncCallback(int64_t aNetworkTimeOffsetDelta)
+{
+    if (mTimeSyncCallback != NULL)
```

Comment:
How about checking the the time sync jump notify here, instead of in ncp_base_mtd.cpp, so the feature could also be used in SoC mode.

Reply:
Furthermore I was thinking that the listener should also be notified when the status changes, regardless of the time jump, for example (pseudocode):
```
if (mTimeSyncCallback != NULL && (ABS(aNetworkTimeOffsetDelta) > OPENTHREAD_CONFIG_NCP_TIME_SYNC_JUMP_NOTIF_MIN_US) || newStatus != oldStatus) {
    Notify();
}
```

Reply:
One thing I did consider though, the `OPENTHREAD_CONFIG_NCP_TIME_SYNC_JUMP_NOTIF_MIN_US` check is to limit the amount of time sync messages sent over the finite bandwidth serial connection to the host. Would an SoC implementation also need such a limit seeing as it's just a function call?

Reply:
Is it designed to notify upper layer every time the timeoffset changes? Or only when it is necessary (time jump, status change). If later, I think we'd better limit the check in time_sync_service.

Reply:
Updated pull request. The check is now performed in time_sync_service.cpp. Also, the callback is called when the status changes.